src/implementations/minecraft_bedrock/server.rs

```diff
@@ -0,0 +1,515 @@
+use std::collections::BTreeMap;
+use std::env;
+use std::process::Stdio;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use color_eyre::eyre::{eyre, Context};
+use sysinfo::{Pid, PidExt, ProcessExt, SystemExt};
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::process::Command;
+
+use crate::error::{Error, ErrorKind};
+use crate::events::{CausedBy, Event, EventInner, InstanceEvent, InstanceEventInner};
+use crate::prelude::LODESTONE_PATH;
+use crate::traits::t_configurable::TConfigurable;
+use crate::traits::t_macro::TMacro;
+use crate::traits::t_server::{MonitorReport, State, StateAction, TServer};
+
+use crate::types::Snowflake;
+use crate::util::dont_spawn_terminal;
+
+use super::MinecraftBedrockInstance;
+use super::player::MinecraftBedrockPlayer;
+use tracing::{debug, error, info, warn};
+
+#[async_trait]
+impl TServer for MinecraftBedrockInstance {
+    async fn start(&mut self, cause_by: CausedBy, block: bool) -> Result<(), Error> {
+        let config = self.config.lock().await.clone();
+        self.state.lock().await.try_transition(
+            StateAction::UserStart,
+            Some(&|state| {
+                self.event_broadcaster.send(Event {
+                    event_inner: EventInner::InstanceEvent(InstanceEvent {
+                        instance_name: config.name.clone(),
+                        instance_uuid: self.uuid.clone(),
+                        instance_event_inner: InstanceEventInner::StateTransition { to: state },
+                    }),
+                    snowflake: Snowflake::default(),
+                    details: "Starting server".to_string(),
+                    caused_by: cause_by.clone(),
+                });
+            }),
+        )?;
+
+        if !port_scanner::local_port_available(config.port as u16) {
+            return Err(Error {
+                kind: ErrorKind::Internal,
+                source: eyre!("Port {} is already in use", config.port),
+            });
+        }
+
+        env::set_current_dir(&self.path_to_instance).context(
+            "Failed to set current directory to the instance's path, is the path valid?",
+        )?;
```

Comment:
Use `.current_dir` on the spawn command instead. This will set the working directory of the ENTIRE lodestone core